File: booking-service/src/main/java/dev/project/booking/integration/kafka/event/BookingExpiredEvent.java
package dev.project.booking.integration.kafka.event;

import java.time.Instant;
import java.util.UUID;

public record BookingExpiredEvent(
        UUID eventId,
        String eventType,
        int eventVersion,
        Instant occurredAt,
        String producer,
        String aggregateType,
        UUID aggregateId,
        UUID correlationId,
        Payload payload
) {
    public record Payload(
            UUID bookingId,
            Instant expiredAt
    ) {
    }
}

File: payment-service/internal/kafka/booking_expired_event.go
package kafka

import (
	"time"

	"github.com/google/uuid"
)

const BookingExpiredEventType = "booking.expired"

type BookingExpiredEvent struct {
	EventID       uuid.UUID                  `json:"eventId"`
	EventType     string                     `json:"eventType"`
	EventVersion  int                        `json:"eventVersion"`
	OccurredAt    time.Time                  `json:"occurredAt"`
	Producer      string                     `json:"producer"`
	AggregateType string                     `json:"aggregateType"`
	AggregateID   uuid.UUID                  `json:"aggregateId"`
	CorrelationID uuid.UUID                  `json:"correlationId"`
	Payload       BookingExpiredEventPayload `json:"payload"`
}

type BookingExpiredEventPayload struct {
	BookingID uuid.UUID `json:"bookingId"`
	ExpiredAt time.Time `json:"expiredAt"`
}
